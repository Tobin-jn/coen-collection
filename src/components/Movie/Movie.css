.Movie {
  background-color: black;
  border-radius: 40px;
  height: 390px;
  margin: 16px;
  width: 260px;
}

.movie-details,
.movie-poster {
  border-radius: 30px;
  cursor: pointer;
  height: 390;
  opacity: 0.9;
  width: 260px;
}

.movie-details {
  background-image: linear-gradient(silver, white);
  height: 390px;
  line-height: 1.2;
  text-align: center;
}

.movie-details:hover,
.movie-poster:hover {
  bottom: 10px;
  display: block;
  height: 400px;
  opacity: 1;
  position: relative;
  right: 10px;
  width: 280px;
}

.movie-details:hover {
  opacity: 0.9;
  overflow: scroll;
}

.movie-date {
  margin: 10px;
}

.text-wrapper {
  margin: auto;
  padding: 20px 0;
  width: 230px;
}
