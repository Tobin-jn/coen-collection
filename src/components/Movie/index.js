import React from 'react';
import './Movie.css'

const Movie = () => {
  return (
    <div className="Movie">Movie</div>
  )
}

export default Movie