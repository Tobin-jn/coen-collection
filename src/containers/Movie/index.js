import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Movie.css";
import { connect } from "react-redux";
import { toggleFavorite, setErrorMessage } from "../../actions";
import * as fetch from "../../utilities/fetch.js";

export class Movie extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isHovered: false
    };
  }

  handleClickFavorite = () => {
    const {
      favorited,
      id,
      handleToggleFavorite,
      handleErrorMessage
    } = this.props;
    if (!this.verifyUserIsLoggedIn()) {
      handleErrorMessage("Please log in or sign up to select favorites");
      return;
    }

    handleToggleFavorite(id);
    if (!favorited) {
      try {
        fetch.fetchAddFavorite(this.props);
      } catch (error) {}
    } else {
      fetch.removeFavorite(this.props.currentUser.id, id);
    }
  };

  handleHover = bool => {
    const isHovered = bool;
    this.setState({ isHovered });
  };

  verifyUserIsLoggedIn = () => {
    return this.props.loggedIn;
  };

  render() {
    let { title, overview, date, poster, favorited } = this.props;
    let cardContents;

    let year = "/" + date.split("-")[0];
    date = date
      .split("-")
      .slice(1)
      .join("/");

    if (!this.state.isHovered) {
      cardContents = (
        <div>
          <i className={"fa-star" + (favorited ? " fas favorited" : " far")} />
          <img className="movie-poster" src={poster} alt="Movie Poster" />
        </div>
      );
    } else {
      cardContents = (
        <div className="movie-details">
          <article className="text-wrapper">
            <i
              className={"fa-star" + (favorited ? " fas favorited" : " far")}
              onClick={this.handleClickFavorite}
            />
            <h1 className="movie-title">{title}</h1>
            <p className="movie-date">
              {date}
              {year}
            </p>
            <p className="movie-overview">{overview}</p>
          </article>
        </div>
      );
    }
    return (
      <div
        onMouseLeave={() => this.handleHover(false)}
        onMouseEnter={() => this.handleHover(true)}
        className="Movie"
      >
        {cardContents}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  currentUser: state.currentUser,
  loggedIn: state.loggedIn
});

export const mapDispatchToProps = dispatch => ({
  handleToggleFavorite: id => dispatch(toggleFavorite(id)),
  handleErrorMessage: message => dispatch(setErrorMessage(message))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Movie);

Movie.propTypes = {
  title: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired,
  voteAverage: PropTypes.number.isRequired,
  poster: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  favorited: PropTypes.bool.isRequired,
  currentUser: PropTypes.object.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  handleToggleFavorite: PropTypes.func.isRequired,
  handleErrorMessage: PropTypes.func.isRequired
};
