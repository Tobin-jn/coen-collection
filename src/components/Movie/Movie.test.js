/* eslint-disable */

import React from "react";
import { shallow } from "enzyme";
import Movie from "./index";

describe("Card", () => {
  it("Renders like snapshot", () => {
    const wrapper = shallow(<Movie />);
    expect(wrapper).toMatchSnapshot();
  });
});
