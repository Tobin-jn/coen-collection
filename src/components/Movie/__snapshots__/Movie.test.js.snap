// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Renders like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="movie-title"
        />,
        <p
          className="movie-overview"
        />,
        <p
          className="movie-date"
        />,
        <img
          alt="Movie Poster"
          className="movie-poster"
        />,
      ],
      "className": "Movie",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "movie-title",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "movie-overview",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "movie-date",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Movie Poster",
          "className": "movie-poster",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="movie-title"
          />,
          <p
            className="movie-overview"
          />,
          <p
            className="movie-date"
          />,
          <img
            alt="Movie Poster"
            className="movie-poster"
          />,
        ],
        "className": "Movie",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "movie-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "movie-overview",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "movie-date",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Movie Poster",
            "className": "movie-poster",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
